### ISSUE TITLE
Session crashes again and again for unknown reason

### ISSUE THREAD

**Issue by yinliu13** [googlecolab/colabtools#3475]
**Describe the current behavior**
Session crashes again and again for unknown reason. I'm using colab pro with gpu. Happens also when I use premium gpu and high ram

**What web browser you are using**
(Chrome)

**Additional context**
These are the app.logs I get when it crashes:
![image](https://user-images.githubusercontent.com/61841209/224165217-f4eb9551-387f-4511-a497-72c6a52b5da2.png)

**Comment by EvanWiederspan** [comment]
Can you share a minimal repro notebook, and which line of code makes the session crash?

**Comment by yinliu13** [comment]
This is the notebook link, https://colab.research.google.com/drive/1128Xs60x3Xzl1UvTksFWE_IX2HN5UAA5?usp=sharing
The lines are at the end of the notebook:
analyzer = innvestigate.create_analyzer("lrp.epsilon", model)
a = analyzer.analyze(img_resized)

I set the recursion limit to 50,000 because otherwise the max depth is reached running this code.

**Comment by EvanWiederspan** [comment]
I see in your 'Learning rate' cell that you're using 'keras.optimizers.Adam'. You may want to take a look at https://github.com/tensorflow/tensorflow/releases/tag/v2.11.0. TF 2.11 introduced some changes to how optimizers are set up. That page recommends switching to the legacy optimizers (`tf.keras.optimizer.legacy.Adam`) if you see issues

If that doesn't change anything, then you may just be running out of RAM, which we can't really fix
